### ISSUE TITLE
Lava Orc feedback, philosophy, and fixes

### ISSUE THREAD

**Issue by yrmvgh** [jeremygurr/dcssca#239]
This race feels like hill orc, but more interesting. So my first lava orc suggestion is... delete hill orcs. The aptitiudes are extremely similar, and both races can do the Beogh thing.

**Comment by yrmvgh** [comment]
Limited control over your temperature might be nice. Maybe when you're at max temp, you can "vent" the heat by exploding in radius 2, which does some ok damage and resets you to minimum temperature.

**Comment by yrmvgh** [comment]
I'm ally-free right now, but your passive fire aura should probably not anger allies, if it currently does. Maybe it could even be a smart aura that refuses to burn allies... I don't know how in the world that would "make sense" thematically but without it, lava orcs might as well act as if they've Sacrificed Love to Ru.

**Comment by yrmvgh** [comment]
> For Lava Orcs, orcish armour impedes spellcasting less than other armours and improves their effective Armour skill. They deal slightly more damage with orcish weapons.

Obviously that will no longer work. Perhaps instead they could have an affinity for (aka improved damage with) the Fire weapon brand. Armor-wise, they could have a bit of an inversion of the Ozocubu limitation. (reduced spell failure chance but only when wearing "heavy" armor)

**Comment by yrmvgh** [comment]
> Any adjacent monster that attacks you in melee will suffer 0-4 points of fire damage (Ignores AC and EV)

Ignoring AC and EV seems to be a bit of a no-no these days in crawl. Instead of ignoring it completely, perhaps it would be better for AC and EV to be halved, or thirded, against the fire aura & fire riposte? But not if that translates into "after reaching (point in the game), lava orc fire stuff no longer does anything."

**Comment by yrmvgh** [comment]
Unfortunately the wiki doesn't make it super clear **why** lava orcs were axed, which makes it harder to guess what is most in need of change.

Re: our recent IRC discussions about vampire simplification, perhaps there are too many slightly-different temperature thresholds on LO, and it could lose one or two of them so that the differences are clearer and easier to remember.

A visible buff to remind you what temperature you're at could also help, after the relevant combat message has scrolled past? Personally I find it pretty easy to keep track of, though.

**Comment by Sandman25DCSS** [comment]
I wouldn't like HO removed, it is pretty unique race.

**Comment by jeremygurr** [comment]
what makes HO unique? Their apts?

**Comment by Sandman25DCSS** [comment]
Yes, aptitudes. Also Beough.

**Comment by yrmvgh** [comment]
Diff between HO and LO

`Fighting   0   Armour      0   Spellcasting    0
Short Blades    0   Dodging     0   Conjurations    0
Long Blades     0   Stealth     0   Hexes   0
Axes    -1      Shields     0   Charms  0
Maces & Flails  0               Summonings  0
Polearms    0               Necromancy  0
Staves  0                   Translocations  0
Unarmed Combat  0               Transmutation   6
            Fire Magic  1       
Throwing    0   Ice Magic   -3  Invocations     0
Slings      -1  Air Magic   -1  Evocations  1
Bows        -1  Earth Magic     1       
Crossbows   -1  Poison Magic    0   Experience  -1
`

**Comment by Sandman25DCSS** [comment]
> Diff between HO and LO

LO are really different from HO due to non-aptitude mechanic. LO cannot replace HO.

**Comment by jeremygurr** [comment]
Yeah the LO mechanic is quite different than the HO. But I don't see HO being all that different from other species, except for Beogh. 

Outside of a beogh gameplay style (which is quite unique and fun), is there any particular playstyle that a HO could pull off that another species couldn't also easily pull off? I guess they are basically a normal-size Str based race, as opposed to elf which is more Int/dex based, ogre which is a large str based race, and human which is balanced. I'm going to eventually make attribute progression exponential instead of linear, which will make this difference a bigger deal (5 -> 10 doubles damage, 10 -> 15 doubles damage, 15 -> 20 doubles damage).

**Comment by yrmvgh** [comment]
With lava orcs around, if beogh also accepted minotaurs, we definitely wouldn't need hill orcs :grinning:

**Comment by Sandman25DCSS** [comment]
Hill Orc is probably the only species who can cast high level spells in heavy armour in 3 rune game. At least it's the best for that role.

**Comment by Sandman25DCSS** [comment]
Minotaurs are quite different from Hill Orc. Hill Orc have good magic aptitudes

**Comment by yrmvgh** [comment]
Well, the reason I'm bandying about races to remove, is that Gurr wants to make room flr faerie, nightstalker, and jelly races. If you love hill orcs we should probably keep hill orcs... I was just thinking that maybe lava orcs can take the place of magical/hybrid hill orcs, and minotaurs can take the place of mundane ones, and between the two maybe we wouldn't need hill orcs any more.

Another possibility we bandied about was elves. DEs and HEs are both spellcasty elves, with one having higher stats and the other having higher apts. Maybe one could go!

**Comment by Sandman25DCSS** [comment]
Well, I think Sledge Elves lost their meaning after changes to Transmutations. Also Humans are like boring Demonspawn who can worship good gods.

**Comment by jeremygurr** [comment]
Humans have very fast levelling and can switch gods with half as much wrath. And they have good apts in both spellcasting and fighting, which is quite rare. And they have a lot more crosstraining going on. By far the most adaptable race. 

And sludge elves have a very unique style of being able to cast high level spells quickly, but not be able to have much spell power. You think they should have higher transmutation apt? It may require more testing to get it right, but I think there's a lot of potential for a very different game style here. 

But I'm okay with creating more species beyond the 30 if all of these really have their place. But that isn't super high priority for me, other changes will come sooner.

**Comment by Sandman25DCSS** [comment]
Oops, I forgot Humans were changed in the fork, they are no longer boring but probably OP instead :-)

The best school for high level-low power spells were Transmutations and Summoning, Sledge Elf were officially the best species for Tm. Now I don't see how to use them, going for Controlled Blink (and Chei) maybe.

Jeremy fixes boring species by making them interesting so I don't think we can easily find a specie for removal, adding new ones is the way to go IMHO.

**Comment by jeremygurr** [comment]
Yeah humans may well be OP. Might be the best beginner race, or for the tinkerers out there that want to be able to try anything they come across in the dungeon. 

I'm not sure I see why SE would not be decent at Tm. Even at low power levels most transmutation spells are still quite powerful, especially if you can get a high powered one early on. Blade hands is completely independent of spell power, as is spider form. But even pulling off an ice beast early on with minimal power is still quite effective probably through mid game. I guess your concern is about lack of late game options for a transmuter. I guess if you focussed on it, blade hands could do quite well even into late game, since it scales with str/dex and fully utilizes UC.

**Comment by Sandman25DCSS** [comment]
SETm have all starting spells (yes, including Blade Hands) at 1% right on turn 0. This can be called decent I guess. Max MP is more limiting factor than anything else here. Of course it feels almost like cheating to me.

**Comment by yrmvgh** [comment]
Silly idea number 335. HO and LO are very similar apt-wise. Help differentiate them - and simultaneously throw a bone to people who miss Dwarves - by nudging one of the orcish races to be a smidge more dwarfy, at least in apts.

**Comment by yrmvgh** [comment]
would be cool if the temp gain correlated (not necessarily on a 1 for 1 basis, but proportionally) to the number of !!! in the combat log.

**Comment by jeremygurr** [comment]
not sure why you opened this one. Didn't you want to change something specific about LO? I can't remember what it was though. Note it here and then I can try and get to it.

**Comment by yrmvgh** [comment]
Temperature gain should correlate to damage, not tension

**Comment by jeremygurr** [comment]
You know, we could make LO gain heat more slowly and lose it more quickly in careful mode, and the opposite in power mode. That gives the player a lot more control over it. If we want more control over it, that is.

**Comment by yrmvgh** [comment]
I really don't like tying it to modes explicitly. Modes are still very subject to change.

If the heat gain / loss is correlated to amount of damage, then it will naturally go up a bit faster in power mode, without relying on an explicit connection to an abstract game mechanic.

**Comment by yrmvgh** [comment]
For the record, here is how I would do heat gain if I had the coding chops.

(reminder: max heat is 15)
- _heat decay_: IF no enemies visible AND no damage dealt/received for 20 aut, THEN heat approaches 0 asymptotically
- _combat is exciting_: IF enemies visible AND heat < 3 , THEN heat approaches 3 linearly
- otherwise, _heat gain_ = MAX(suffer, deal, burn)

**suffer**:
- IF player is damaged this "turn" (within past 10 aut), THEN heat goes up by a number equal to the heaviest hit in !!!, plus 1 more heat if there were multiple attacks suffered.
  - eg, you are hit by a jackal!! and stung by a bee! x2. You get 2 heat for the jackal hit plus 1 for the presence of other hits.

**deal**:
- the highest number of !!! the player dealt this "turn" to any single opponent

**burn**:
- if in lava, go to max heat. if in fire cloud, heat +3. if suffering sticky flame, heat + 1.
- as implied by following the above rules, heat does not decay in combat. that feels wrong and silly. as such, "[lava vent](https://github.com/yrmvgh/dcssca/blob/master/crawl-ref/yrv-staging/lava-vent.cc)" would probably need to exist, too.

**Comment by yrmvgh** [comment]
That's probably way overcomplicating heat gain, but that's what I would do given infinite skill, and infinite time to work out bugs and write all these new functions and such.

**Comment by jeremygurr** [comment]
I'm not saying the power/care mode effects would replace the damage part, it would be in addition to it.

**Comment by yrmvgh** [comment]
Hrrrrrm. I still don't like it. I dislike it a lot. Trying to figure out why...

I think that I dislike it because it's stapling two somewhat unrelated things together. You want to try to hang onto your armor buff, but you're not actually interested in +acc +stealth +spell and -stamina? Tough, it's a bundled cable TV package and you can't stop the kids from watching the extra channels that you've been given!

Want to pick a mode to help raise/lower your heat but you're out of stamina? Maybe because you berserked? Well, otther races' racial mechanics aren't bundled into the Mode system, but yours are, so you'll have to wait! It can be flavor justified (you're too tired to control your heat!) but mechanically I just don't think it's a good fit.

Ultimately though, I understand you have a looooot of coding to do, so if you choose to take on this heat thing, by all means code it how you wish! Just lodging my opinions for the record. :smile_cat: But this is _somewhere_ on my want-to-do list, so don't feel like you have to do it.

**Comment by jeremygurr** [comment]
Ok no mode effects for LO :)
